Add ARM64 to release.sh

namazso opened the pull request:
gcc doesn't support targetting windows on arm64 yet.
I couldn't get the other gcc ones build either on my machine, but shouldn't matter since I didn't change anything there.

namazso commented:
If you have suggestions on changing the folder name or anything, feel free to comment or straight up edit and push onto the branch

Also, not sure if this is intended or if I should open an issue for this, but using the `zip` msys2 package as zipper results in `tmp` being in the root of the output zip file.

![image](https://user-images.githubusercontent.com/8676443/78072551-9b41da80-739f-11ea-8310-7f97fa25c08d.png)

mity commented:
> Also, not sure if this is intended or if I should open an issue for this, but using the zip msys2 package as zipper results in tmp being in the root of the output zip file.

It's not. On my machine I have 7za in the path which behaves differently. Maybe the fallback to zip was not tested.

mity commented:
I have committed some fixes. Now, after installing ARM64 support into MSVC 2019 it starts to work.

Except the build of one example fails though (just for this platform). But that can be solved separately I guess.

<details>
    <summary>The build log</summary>

```
1>------ Build started: Project: ZERO_CHECK, Configuration: Debug ARM64 ------
1>Checking Build System
2>------ Build started: Project: hsluv-c, Configuration: Debug ARM64 ------
3>------ Build started: Project: md4c-utf16, Configuration: Debug ARM64 ------
4>------ Build started: Project: example-dllgetversion, Configuration: Debug ARM64 ------
5>------ Build started: Project: test-build, Configuration: Debug ARM64 ------
2>Building Custom Rule D:/prj/mctrl.namazso/lib/hsluv-c/src/CMakeLists.txt
3>Building Custom Rule D:/prj/mctrl.namazso/lib/md4c-utf16/CMakeLists.txt
2>hsluv.c
4>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
5>Building Custom Rule D:/prj/mctrl.namazso/tests/CMakeLists.txt
3>md4c-utf16.c
2>hsluv-c.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\hsluv-c.lib
3>D:\prj\mctrl.namazso\lib\md4c-utf16\../md4c/md4c/md4c.c(908,35): warning C4244: '=': conversion from '__int64' to 'MD_SIZE', possible loss of data
3>D:\prj\mctrl.namazso\lib\md4c-utf16\../md4c/md4c/md4c.c(1585,21): warning C4018: '>=': signed/unsigned mismatch
3>D:\prj\mctrl.namazso\lib\md4c-utf16\../md4c/md4c/md4c.c(1590,21): warning C4018: '>=': signed/unsigned mismatch
3>D:\prj\mctrl.namazso\lib\md4c-utf16\../md4c/md4c/md4c.c(2224,1): warning C4244: 'function': conversion from '__int64' to 'int', possible loss of data
3>D:\prj\mctrl.namazso\lib\md4c-utf16\../md4c/md4c/md4c.c(4215,1): warning C4244: 'function': conversion from '__int64' to 'int', possible loss of data
3>D:\prj\mctrl.namazso\lib\md4c-utf16\../md4c/md4c/md4c.c(4846,1): warning C4244: 'initializing': conversion from '__int64' to 'MD_OFFSET', possible loss of data
3>md4c-utf16.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\md4c-utf16.lib
4>example-dllgetversion.c
5>test-build.c
3>Done building project "md4c-utf16.vcxproj".
6>------ Build started: Project: mCtrl, Configuration: Debug ARM64 ------
6>Building Custom Rule D:/prj/mctrl.namazso/src/CMakeLists.txt
5>test-build.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\test-build.exe
4>example-dllgetversion.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-dllgetversion.exe
6>buffer.c
6>memstream.c
6>anim.c
6>button.c
6>chart.c
6>color.c
6>comua.c
6>debug.c
6>dialog.c
6>doublebuffer.c
6>dsa.c
6>entity.c
6>expand.c
6>generic.c
6>grid.c
6>html.c
6>imgview.c
6>labeledit.c
6>mditab.c
6>mdtext.c
6>Generating Code...
6>Compiling...
6>mdview.c
6>menubar.c
6>misc.c
6>module.c
6>mousedrag.c
6>mousewheel.c
6>rgn16.c
6>table.c
6>tooltip.c
6>treelist.c
6>url.c
6>version.c
6>viewlist.c
6>xcom.c
6>xdwm.c
6>xd2d.c
6>xdwrite.c
6>xwic.c
6>Generating Code...
6>   Creating library D:/prj/mctrl.namazso/xx/Debug/mCtrl.lib and object D:/prj/mctrl.namazso/xx/Debug/mCtrl.exp
6>mCtrl.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\mCtrl.dll
7>------ Build started: Project: example-button, Configuration: Debug ARM64 ------
8>------ Build started: Project: example-chart, Configuration: Debug ARM64 ------
9>------ Build started: Project: example-expand, Configuration: Debug ARM64 ------
10>------ Build started: Project: example-grid, Configuration: Debug ARM64 ------
9>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
8>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
10>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
7>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
9>example-expand.c
10>example-grid.c
7>example-button.c
8>example-chart.c
10>example-grid.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-grid.exe
11>------ Build started: Project: example-grid-customdraw, Configuration: Debug ARM64 ------
7>example-button.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-button.exe
9>example-expand.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-expand.exe
8>example-chart.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-chart.exe
12>------ Build started: Project: example-grid-ownerdata, Configuration: Debug ARM64 ------
13>------ Build started: Project: example-html, Configuration: Debug ARM64 ------
14>------ Build started: Project: example-imgview, Configuration: Debug ARM64 ------
15>------ Build started: Project: example-mditab, Configuration: Debug ARM64 ------
13>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
12>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
15>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
14>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
12>example-grid-ownerdata.c
15>example-mditab.c
14>example-imgview.c
15>D:\prj\mctrl.namazso\examples\example-mditab.c(66,59): warning C4244: '=': conversion from 'LRESULT' to 'int', possible loss of data
15>D:\prj\mctrl.namazso\examples\example-mditab.c(67,79): warning C4244: '=': conversion from 'LRESULT' to 'int', possible loss of data
13>example-html.c
12>example-grid-ownerdata.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-grid-ownerdata.exe
16>------ Build started: Project: example-mditab-dwm, Configuration: Debug ARM64 ------
13>example-html.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-html.exe
17>------ Build started: Project: example-mdview, Configuration: Debug ARM64 ------
15>example-mditab.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-mditab.exe
15>Done building project "example-mditab.vcxproj".
18>------ Build started: Project: example-menubar, Configuration: Debug ARM64 ------
16>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
17>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
18>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
16>example-mditab-dwm.c
18>example-menubar.c
16>D:\prj\mctrl.namazso\examples\example-mditab-dwm.c(68,59): warning C4244: '=': conversion from 'LRESULT' to 'int', possible loss of data
17>example-mdview.c
16>D:\prj\mctrl.namazso\examples\example-mditab-dwm.c(69,79): warning C4244: '=': conversion from 'LRESULT' to 'int', possible loss of data
14>example-imgview.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-imgview.exe
19>------ Build started: Project: example-treelist, Configuration: Debug ARM64 ------
19>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
18>example-menubar.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-menubar.exe
16>example-mditab-dwm.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-mditab-dwm.exe
19>example-treelist.c
16>Done building project "example-mditab-dwm.vcxproj".
20>------ Build started: Project: example-treelist-ownerdata, Configuration: Debug ARM64 ------
17>example-mdview.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-mdview.exe
21>------ Build started: Project: test-grid, Configuration: Debug ARM64 ------
22>------ Build started: Project: test-table, Configuration: Debug ARM64 ------
20>Building Custom Rule D:/prj/mctrl.namazso/examples/CMakeLists.txt
21>Building Custom Rule D:/prj/mctrl.namazso/tests/CMakeLists.txt
22>Building Custom Rule D:/prj/mctrl.namazso/tests/CMakeLists.txt
20>example-treelist-ownerdata.c
22>test-table.c
21>test-grid.c
20>D:\prj\mctrl.namazso\examples\example-treelist-ownerdata.c(132,49): warning C4267: '+=': conversion from 'size_t' to 'DWORD', possible loss of data
19>example-treelist.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\example-treelist.exe
20>example-treelist-ownerdata.obj : error LNK2019: unresolved external symbol __imp_RegCloseKey referenced in function OnTreeListNotify
20>example-treelist-ownerdata.obj : error LNK2019: unresolved external symbol __imp_RegEnumKeyExW referenced in function InsertChildren
20>example-treelist-ownerdata.obj : error LNK2019: unresolved external symbol __imp_RegEnumValueW referenced in function InsertChildren
20>example-treelist-ownerdata.obj : error LNK2019: unresolved external symbol __imp_RegOpenKeyExW referenced in function InsertChildren
20>example-treelist-ownerdata.obj : error LNK2019: unresolved external symbol __imp_RegQueryInfoKeyW referenced in function KeyHasChildren
20>D:\prj\mctrl.namazso\xx\Debug\example-treelist-ownerdata.exe : fatal error LNK1120: 5 unresolved externals
20>Done building project "example-treelist-ownerdata.vcxproj" -- FAILED.
21>test-grid.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\test-grid.exe
22>test-table.vcxproj -> D:\prj\mctrl.namazso\xx\Debug\test-table.exe
23>------ Skipped Build: Project: ALL_BUILD, Configuration: Debug ARM64 ------
23>Project not selected to build for this solution configuration 
========== Build: 21 succeeded, 1 failed, 0 up-to-date, 1 skipped ==========
```
</details>

mity commented:
> gcc doesn't support targetting windows on arm64 yet.

I believe there is some ongoing work in mingw-w64 but IDK current status. It's possible it might require an extra gcc-toolchain to be installed. But it's not a showstopper to me. We can add that anytime later when needed or when it reaches some maturity.

mity commented:
From my POV, it's ready for merging. 

I'm not sure whether you want to fix the one example failing in the ARM64 build before doing so or treat it as a separate thing.

namazso commented:
I also noticed that problem myself, that's why I added the advapi32 linking to example-grid-ownerdata (incorrectly, since the failing one was example-treelist-ownerdata). On my VM it all built fine though, so I assumed my fix was correct.
As it turns out it wasn't, and it just worked via some kind of dark magic. Anyways, I fixed that correctly this time

mity commented:
Merged. Thanks.
